On Ubuntu 20.04, which has Python 3.8, I had installed Python 3.10 from the deadsnakes PPA but I got this error running `stbt_rig.py setup`:

```
❯ ./stbt_rig.py setup --vscode
stbt_rig.py: WARNING: Virtualenv in '/home/drothlis/work/stb-tester.com/test-packs/demo/.venv' has Python version 3.6.15; deleting .venv and recreating it with Python 3.10.
Error: Command '['/home/drothlis/work/stb-tester.com/test-packs/demo/.venv/bin/python3.10', '-Im', 'ensurepip', '--upgrade', '--default-pip']' returned non-zero exit status 1.
Traceback (most recent call last):
  File "./stbt_rig.py", line 1925, in <module>
    sys.exit(main())
  File "./stbt_rig.py", line 83, in main
    return main_with_args(args)
  File "./stbt_rig.py", line 110, in main_with_args
    return cmd_setup(args, args.node_id)
  File "./stbt_rig.py", line 544, in cmd_setup
    setup_stage1(this_stbt_rig, root)
  File "./stbt_rig.py", line 584, in setup_stage1
    cwd=root)
  File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '[u'python3.10', u'-m', u'venv', u'.venv']' returned non-zero exit status 1
```

The solution was to install `python3.10-venv` (see https://stackoverflow.com/a/71726397).

--- a/stbt_rig.py
+++ b/stbt_rig.py
@@ -576,7 +576,13 @@ def setup_stage1(this_stbt_rig, root):
                 "Virtualenv in '%s' has Python version %s; "
                 "deleting .venv and recreating it with Python %s.",
                 venv_dir, venv_python_version, v)
-            shutil.rmtree(venv_dir)
+            while True:
+                ok = ask("Delete and recreate .venv? [y/n] ").lower().strip()
+                if ok == "y":
+                    shutil.rmtree(venv_dir)
+                    break
+                elif ok == "n":
+                    break
 
     if not os.path.exists(venv_dir):
         # Create .venv

